### ISSUE TITLE
Parent child relationship in embedded models

### ISSUE THREAD

**Issue by mitsos1os** [strongloop/loopback-datasource-juggler#1784]
Hi there, I understand that Loopback 3 is in LTS mode and no new feature will be implemented. However if you approve this, I am willing to issue a pull request to implement this, so I just want to know if you are willing to accept it

## Suggestion

My suggestion is that when embedding model instances in an other model's properties, a meta property is added that will reference the parent of the child model.

## Use Cases

This would be usefull when passing around model instances that need to access their *containing* context for some functionality. Right now having an embedded model instance has no relation to the parent model that contains it.

## Examples

Let's say we have a **ChildModel** such as this:
```js
{
	name: 'ChildModel',
	properties: {
		name: 'string',
		notes: 'string'
	}
}
```
and a **ParentModel**:
```js
{
	name: 'ParentModel',
	properties: {
		name: 'string',
		child: 'ChildModel'
	}
}
```
which contains a child property of the type ChildModel. When instantiating the the model instance in a way such as this:
```js
const parent = new ParentModel({
	name: 'John',
	child: {
		name: 'John\'s kid' ,
		notes: 'Good Boy'
	}
})
```
the `parent.child` should contain a property (such as *`__parent`*) that references the parent model. ex:
`parent.child.__parent === parent`

## Acceptance criteria

TBD - will be filled by the team.

**Comment by dhmlau** [comment]
@mitsos1os, are you looking for this feature being implemented in `3.x` or `4.x`?  `4.x` is still being used in LB4, so we're accepting new features.  

@raymondfeng @bajtos, what's your thought on this? Thanks.

**Comment by mitsos1os** [comment]
@dhmlau I was going to implement it in the 3.x branch. But I am open to your suggestions

**Comment by mitsos1os** [comment]
> @dhmlau I was going to implement it in the 3.x branch. But I am open to your suggestions

I checked it and the code changes seem pretty straightforward. So if you are willing to accept it for the 3.x branch, I could implement it for the 4.x latest branch as well

**Comment by raymondfeng** [comment]
@mitsos1os Your proposal seems to be reasonable. Please make sure the `__parent` property won't interfere with persistence.

**Comment by mitsos1os** [comment]
@raymondfeng Thank you. I will submit the pull request for you guys to check.

**Comment by mitsos1os** [comment]
I have already issued a pull request #1787 and waiting for it to move on. Please don't mark as stale

**Comment by mitsos1os** [comment]
Ongoing discussion on related pull request. Issue not stale

**Comment by mitsos1os** [comment]
Resolved through #1834
